test/UniversalRouterDistributor.t.sol

```diff
@@ -151,13 +156,25 @@ contract UniversalRouterDistributor is Test {
             uint256 balanceBefore1 = ERC20(address(token1)).balanceOf(vm.addr(index));
             uint256 balanceBefore2 = ERC20(address(token2)).balanceOf(vm.addr(index));
 
+            // Claim token1
+            vm.expectEmit(true, true, true, true, address(distributor));
+            emit RewardsClaimed(vm.addr(index), address(token1), claimableAdjusted1);
             distributor.claim(vm.addr(index), address(token1), claimableInput, proof1);
+            // Claim token2
+            vm.expectEmit(true, true, true, true, address(distributor));
+            emit RewardsClaimed(vm.addr(index), address(token2), claimableAdjusted2);
             distributor.claim(vm.addr(index), address(token2), claimableInput, proof2);
 
+            uint256 balanceAfter1 = balanceBefore1 + claimableAdjusted1;
+            uint256 balanceAfter2 = balanceBefore2 + claimableAdjusted2;
+
             assertEq(ERC20(address(token1)).balanceOf(address(distributor)), 0);
-            assertEq(ERC20(address(token1)).balanceOf(vm.addr(index)), balanceBefore1 + claimableAdjusted1);
+            assertEq(ERC20(address(token1)).balanceOf(vm.addr(index)), balanceAfter1);
             assertEq(ERC20(address(token2)).balanceOf(address(distributor)), 0);
-            assertEq(ERC20(address(token2)).balanceOf(vm.addr(index)), balanceBefore2 + claimableAdjusted2);
+            assertEq(ERC20(address(token2)).balanceOf(vm.addr(index)), balanceAfter2);
+            // Assert claimed getter
+            assertEq(distributor.claimed(vm.addr(index), address(token1)), balanceAfter1);
+            assertEq(distributor.claimed(vm.addr(index), address(token2)), balanceAfter2);
```

Comment:
Regarding the test of the `claimed` getter, I would say those asserts are not sufficient. `balanceAfterX` depends on `claimableAdjustedX` which itself depends on `distributor.claimed` implementation.
I propose to change
```solidity
uint256 claimableAdjusted2 = claimableInput - distributor.claimed(vm.addr(index), address(token2));
```
to 
 ```solidity
uint256 claimableAdjusted2 = claimableInput - vm.load(address(distributor), DATA_SLOT); 
```
With `DATA_SLOT` to calculate according to layout of storage for mapping.
It would ensure the right slot is loaded in the implementation and that the assert really test the implementation.
What are your thoughts on this  @MerlinEgalite ?

Reply:
After local tests and tinkering, it would be 
```solidity
uint256 claimableAdjusted1 = claimableInput - uint256(vm.load(address(distributor), keccak256(abi.encode(address(token1), keccak256(abi.encode(vm.addr(index), uint256(2)))))));
uint256 claimableAdjusted2 = claimableInput - uint256(vm.load(address(distributor), keccak256(abi.encode(address(token2), keccak256(abi.encode(vm.addr(index), uint256(2)))))));
```

Will do a proper refactored commit shortly

Reply:
added a unit test for claimed getter